pkg/pull/pull.go

```diff
@@ -26,7 +28,9 @@ func (f *FilterImpl) Configs() []FilterConfig {
 }
 
 type Puller interface {
```

Comment:
Can we possibly come up with a slightly more self explanatory name for this interface.  `OrasPuller`?

Reply:
i can add that as a TODO and address in a follow-up

Reply:
Right but what does it pull? and how is it different from a fetch for instance.

Reply:
i agree with the Fetcher/Puller confusion. let me take a look and see if i can fix that up